pennylane/ftqc/pauli_tracker.py

```diff
@@ -235,3 +244,353 @@ def commute_clifford_op(clifford_op: Operator, xz: List[Tuple[int, int]]) -> Lis
+
+
+def _parse_cnot(mid_meas: List):
+    """Parse the mid measurements of a :class:`~pennylane.CNOT` gate.
+
+    Args:
+        mid_meas (List): A list of mid measurement results.
+
+    Returns:
+        A list of tuples of xz encoding.
+    """
+    # Control wire
+    by_op = [
+        (
+            _sum_parity(mid_meas[1], mid_meas[2], mid_meas[4], mid_meas[5]),
+            _sum_parity(
+                mid_meas[0],
+                mid_meas[2],
+                mid_meas[3],
+                mid_meas[4],
+                mid_meas[6],
+                mid_meas[7],
+                mid_meas[9],
+                1,
+            ),
+        )
+    ]
+    # Target wire
+    by_op.append(
+        (
+            _sum_parity(
+                mid_meas[1], mid_meas[2], mid_meas[6], mid_meas[8], mid_meas[10], mid_meas[12]
+            ),
+            _sum_parity(mid_meas[7], mid_meas[9], mid_meas[11]),
+        )
+    )
+    return by_op
+
+
+def _parse_rotxzx(mid_meas: List):
+    """Parse the mid measurements of a :class:`~pennylane.ftqc.RotXZX` gate.
+
+    Args:
+        mid_meas (List): A list of mid measurement results.
+
+    Returns:
+        A list of tuples of xz encoding.
+    """
+    return [(mid_meas[1] ^ mid_meas[3], mid_meas[0] ^ mid_meas[2])]
+
+
+def _parse_rotz(mid_meas: List):
+    """Parse the mid measurements of a :class:`~pennylane.RZ` gate.
+
+    Args:
+        mid_meas (List): A list of mid measurement results.
+
+    Returns:
+        A list of tuples of xz encoding.
+    """
+    return [(mid_meas[1] ^ mid_meas[3], mid_meas[0] ^ mid_meas[2])]
+
+
+def _parse_mid_measurements(tape: QuantumScript, mid_meas: List):
+    r"""Parse a serial of mid-measurement results of a quantum tape with a few number of non-Clifford
+    gates (:class:`~pennylane.RZ` and :class:`~pennylane.ftqc.RotXZX`), up to the number of wires of
+    the tape and one non-Clifford gate per wire, at the beginning of the circuit, Pauli operators
+    (:class:`~pennylane.PauliY`, :class:`~pennylane.PauliZ` and :class:`~pennylane.Identity`) and a
+    set of Clifford gates (:class:`~pennylane.Hadamard`, :class:`~pennylane.S`, :class:`~pennylane.CNOT`).
+    Both nob-Clifford and Clifford gates mentioned above are measured in the way defined in
+    `Raussendorf et al. <https://arxiv.org/abs/quant-ph/0301052>`__.
+
+    Args:
+        tape (QuantumScript): The quantum tape in the standard circuit mode (Gates are not transformed into the MBQC formalism).
+        mid_meas (list): Mid-measurements results.
+
+    Returns:
+        A list of `byproduct` ops in xz and a list of `operations` in a reversed manner.
+    """
+    ops = tape.operations
+
+    by_ops = []
+
+    mid_meas_offset = 0
+    _t_ops_record = [None] * tape.num_wires
+    for idx, op in enumerate(ops):
+        gate_offset = 4 if op.num_wires == 1 else 13
+        ms = mid_meas[mid_meas_offset : mid_meas_offset + gate_offset]
+        by_op = []
+        if isinstance(op, S):
```

Comment:
Nothing to be done here, but if we don't need to care about the inheritance structure of the ops, we can use the Python structured pattern matching instead https://docs.python.org/3/tutorial/controlflow.html#tut-match